library/src/google-maps-provider.tsx

```diff
@@ -80,13 +80,49 @@ export const GoogleMapsProvider: React.FunctionComponent<
     });
     /* eslint-enable camelcase */
 
-    scriptTag.type = 'text/javascript';
-    scriptTag.src = `https://maps.googleapis.com/maps/api/js?${params.toString()}`;
-    scriptTag.onload = (): void => {
-      setIsLoadingAPI(false);
-      onLoadScript && onLoadScript();
-    };
-    document.getElementsByTagName('head')[0].appendChild(scriptTag);
+    const existingScriptTag: HTMLScriptElement | null = document.querySelector(
+      `script[src^="${GOOGLE_MAPS_API_URL}"]`
+    );
+
+    // Check if Google Maps API was loaded with the passed parameters
+    if (existingScriptTag) {
+      const loadedURL = new URL(existingScriptTag.src);
+      const loadedParams = loadedURL.searchParams.toString();
+      const passedParams = params.toString();
+
+      if (loadedParams !== passedParams) {
+        console.error(
+          'The Google Maps API Parameters passed to the `GoogleMapsProvider` components do not match. The Google Maps API can only be loaded once. Please make sure to pass the same API parameters to all of your `GoogleMapsProvider` components.',
+          '\n\nExpected parameters:',
+          Object.fromEntries(loadedURL.searchParams),
+          '\n\nReceived parameters:',
+          Object.fromEntries(params)
+        );
+      }
+    }
+
+    if (typeof google === 'object' && typeof google.maps === 'object') {
+      // Google Maps API is already loaded
+      apiLoadingFinished();
+    } else if (existingScriptTag) {
+      // Google Maps API is already loading
+      setIsLoadingAPI(true);
+
+      const originalOnload = existingScriptTag.onload;
```

Comment:
```suggestion
      const previousOnload = existingScriptTag.onload;
```

Reply:
Hmm I see, it's not really the original onload, but it's also not the previous onload. What do just think about just `onload`?

Reply:
Okay, I see... Can't we just say `existingScriptTag.onload.call(...)` in line 113? 

Otherwise `onload` or maybe `currentOnload` should work (IMO).

Reply:
No, this doesn't work. I need to save it outside of the onload event handler, otherwise it would call itself again and again. So I'll just go with `onload`. 🙂

Reply:
Listen and Repeat :D